src/client.rs

```diff
@@ -952,13 +952,15 @@ where
                             snapshot,
                             Some((&meta, delegation.role().clone())),
                         ));
-                    let (term, res) = f.await;
 
-                    if term && res.is_err() {
-                        return (true, res);
+                    match f.await {
+                        (term, res @ Ok(_)) => return (term, res),
```

Comment:
by passing `term` through to the caller, it is now unused.  Something seems off with that.

Reply:
I'll just switch to `(_, res @ Ok(_)) => return (true, res)` so that we explicitly terminate if we ever find our target.